Lib/test/test_property.py

```diff
@@ -214,6 +214,23 @@ def test_property_set_name_incorrect_args(self):
             ):
                 p.__set_name__(*([0] * i))
 
+    def test_property_setname_on_property_subclass(self):
+        # https://github.com/python/cpython/issues/100942
+        # Copy was setting the name field without first
+        # verifying that the copy was an actual property
+        # instance.  As a result, the code below was
+        # causing a segfault.
+
+        class pro(property):
+            def __new__(typ, *args, **kwargs):
+                return "abcdef"
+
+        class A:
+            pass
+
+        p = property.__new__(pro)
+        p.__set_name__(A, 1)
```

Comment:
This line is not really required. Segfault happens and without it.

Reply:
I prefer to leave it because it covers an additional code path that could fail and it is an accurate record of how the issue was discovered.